### ISSUE TITLE
Derive `DirectoryObject` from `Path`?

### ISSUE THREAD

**Issue by samwaseda** [pyiron/pyiron_workflow#218]
Currently [`DirectoryObject`](https://github.com/pyiron/pyiron_workflow/blob/7fcae5f00f9873a1f5e35301c4aedb8ab5e1aaf5/pyiron_workflow/snippets/files.py#L38) is its own class. There are, however, functionalities that are given almost identically in `Path` from `pathlib` (which `DirectoryObject` anyway heavily uses). Furthermore there are functionalities that I would love to have but are not implemented in `DirectoryObject` yet, such as `path_1 / path_2` etc. So I thought maybe it makes more sense to derive `DirectoryObject` from `Path`, so that people can straightforwardly use the same functionalities. What do you think? @liamhuber

**Comment by liamhuber** [comment]
💯 I am huge fan of this. Maybe it will require merging `DirectoryObject` and `FileObject` together.

The biggest catch I can think of will be to ensure that we exclusively _extend_ rather than _modify_ `pathlib.Path`, as I'm sure there's a billion places where people rely on the pattern `if isinstance(thing, pathlib.Path): thing_that_depends_on_Pathlike_behavior(thing)`, and it would be nice if it was safe to simply plug our objects in under these conditions.

**Comment by liamhuber** [comment]
Somewhat related, there has been some activity in `pyiron_base` modifying the working directory behaviour, and it would be lovely if the `snippets` solution could be used across the pyiron-verse. To that end, it would be good to double check what changes were made and why, and see whether we can adapt what's done here to accomplish the same thing.

Similarly, the stuff with browsing zipped content might be a cool feature to add in.

**Comment by samwaseda** [comment]
> Maybe it will require merging `DirectoryObject` and `FileObject` together.

Yeah that's actually my biggest concern. Right now I really like the fact that the files and directories are separated, because that's somewhat crucial that each node has its own working directory and file objects are derived from them, and all this is properly conceptualised.

**Comment by liamhuber** [comment]
It sounds like we're on the same page (albeit your page is more densely packed with knowledge). I guess a reasonable attack is to simply try moving forward with the re-parenting as see how it turns out.
